client.py

```diff
@@ -0,0 +1,141 @@
+import uuid
+import requests
+import json
+import argparse
+from typing import Dict, Optional, Iterator, Any
+
+
+class ZenoClient:
+    def __init__(self, base_url: str = "http://localhost:8000"):
+        """
+        Initialize the Zeno API client.
+        
+        Args:
+            base_url: The base URL of the Zeno API server
+        """
+        self.base_url = base_url
+        
+    def chat(
+        self, 
+        query: str, 
+        user_persona: Optional[str] = None, 
+        thread_id: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+        session_id: Optional[str] = None,
+        user_id: Optional[str] = None,
+        tags: Optional[list] = None,
+    ) -> Iterator[Dict[str, Any]]:
+        """
+        Send a chat request to the Zeno API and stream the responses.
+        
+        Args:
+            query: The query to send
+            user_persona: Optional user persona
+            thread_id: Optional thread ID
+            metadata: Optional metadata
+            
+        Returns:
+            An iterator of response messages
+        """
+        url = f"{self.base_url}/api/chat"
+        
+        payload = {
+            "query": query,
+        }
+        
+        if user_persona:
+            payload["user_persona"] = user_persona
+            
+        if thread_id:
+            payload["thread_id"] = thread_id
+        
+        if metadata:
+            payload["metadata"] = metadata
+        
+        if session_id:
+            payload["session_id"] = session_id
+        
+        if user_id:
+            payload["user_id"] = user_id
+        
+        if tags:
+            payload["tags"] = tags
+        
+        with requests.post(url, json=payload, stream=True) as response:
+            if response.status_code != 200:
+                raise Exception(f"Request failed with status code {response.status_code}: {response.text}")
+            for update in response.iter_lines():
+                if update:
+                    # Decode the line and parse the JSON
+                    decoded_update = update.decode('utf-8')
+                    yield json.loads(decoded_update)
```

Comment:
Do we still think this is a good way of sending json data? Might be worth reconsidering. Not sure if using a proper json output format would make this better.